### PR TITLE
feat: new is typing implementation refs: WSC-1329

### PR DESCRIPTION
**Pull request by yukiLu6**

### CODE REVIEW

**Review by CerDennis — CHANGES_REQUESTED** [review]
Running tests I see some warnings for update states due to timout usages, please fix them

image

File: src/chats/components/conversation/ConversationHeader.test.tsx [hunk 1]
```diff
@@ -63,4 +98,48 @@ describe('Conversation header test', () => {
...
+
+			store.setIsWriting(mockedRoom.id, mockRobertoUser.id, true);
+			store.setIsWriting(mockedRoom.id, mockGianniUser.id, true);
+			store.setIsWriting(mockedRoom.id, mockLucaUser.id, true);
+			store.setIsWriting(mockedRoom.id, mockQuintoUser.id, true);
+		});
+
+		setup(<ConversationHeader roomId={mockedRoom.id} setInfoPanelOpen={jest.fn()} />);
+
+		const isWriting = await screen.findByText(/Roberto and 3 others are typing.../i);
```
Comment (CerDennis) [inline src/chats/components/conversation/ConversationHeader.test.tsx]: Here you are harcoding the name of the user, extract it from the mock object

File: src/chats/components/conversation/ConversationHeader.test.tsx [hunk 2]
```diff
@@ -63,4 +98,48 @@ describe('Conversation header test', () => {
...
+		});
+
+		expect(isWriting).not.toBeVisible();
+	});
+
+	test('is writing label for four or more users that are writing', async () => {
+		const store: RootStore = useStore.getState();
+		act(() => {
+			store.addRoom(mockedRoom);
+			store.setLoginInfo(mockPaoloUser.id, 'Paolo');
```
Comment (CerDennis) [inline src/chats/components/conversation/ConversationHeader.test.tsx]: Here you are harcoding the name of the user, extract it from the mock object

File: src/chats/components/conversation/ConversationHeader.test.tsx [hunk 3]
```diff
@@ -63,4 +98,48 @@ describe('Conversation header test', () => {
 		setup(<ConversationHeader roomId={mockedRoom.id} setInfoPanelOpen={jest.fn()} />);
 		expect(screen.queryByTestId('ConversationHeaderMeetingButton')).not.toBeInTheDocument();
 	});
+
+	test('is writing appears when someone is writing and disappear if not', async () => {
+		const store: RootStore = useStore.getState();
+		act(() => {
+			store.addRoom(mockedRoom);
+			store.setLoginInfo(mockPaoloUser.id, 'Paolo');
```
Comment (CerDennis) [inline src/chats/components/conversation/ConversationHeader.test.tsx]: Here you are harcoding the name of the user, extract it from the mock object

File: src/chats/components/conversation/ConversationHeader.test.tsx [hunk 4]
```diff
@@ -63,4 +98,48 @@ describe('Conversation header test', () => {
 		setup(<ConversationHeader roomId={mockedRoom.id} setInfoPanelOpen={jest.fn()} />);
 		expect(screen.queryByTestId('ConversationHeaderMeetingButton')).not.toBeInTheDocument();
```
Comment (CerDennis) [inline src/chats/components/conversation/ConversationHeader.test.tsx:98]: Split the tests for the is writing function in a separte suite so it's cleaner

File: src/meetings/components/sidebar/MeetingConversationAccordion/MeetingConversationAccordion.tsx
Comment (CerDennis) [inline src/meetings/components/sidebar/MeetingConversationAccordion/MeetingConversationAccordion.tsx:1]: Here we could also split in a new component the iswriting managment make it more atomic

File: src/hooks/useIsWritingLabel.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	isInsideSidebar?: boolean
+): string | undefined => {
+	const [t] = useTranslation();
+	const isTypingLabel = t('status.isTyping', 'is typing...');
+	const areTypingLabel = t('status.areTyping', 'are typing...');
+
+	const roomType = useStore((store) => getRoomTypeSelector(store, roomId));
+	const writingListNames = useStore((state) => getRoomIsWritingList(state, roomId));
+
+	if (writingListNames === undefined || writingListNames?.length === 0) return undefined;
```
Comment (CerDennis) [inline src/hooks/useIsWritingLabel.tsx:26]: please add an aempty line at the end of a statement :'(

File: src/chats/components/conversation/ConversationHeader.tsx [hunk 6]
```diff
@@ -59,7 +126,21 @@ const ConversationHeader = ({
 			padding={{ vertical: 'medium', horizontal: 'large' }}
 		>
 			<Row takeAvailableSpace mainAlignment="flex-start">
-				<TextWithTooltip overflow="ellipsis">{roomName}</TextWithTooltip>
+				<Container orientation="vertical" width="fit" height=" fit" crossAlignment="flex-start">
+					<CustomTextWithTooltip overflow="ellipsis" $isWritingIsVisible={isWritingIsDefined}>
+						{roomName}
+					</CustomTextWithTooltip>
+					{
```
Comment (lisaparma) [inline src/chats/components/conversation/ConversationHeader.tsx]: `{` and `}` are useless if there isn't a condition

File: src/meetings/components/sidebar/MeetingConversationAccordion/MeetingChatAccordionTitle.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+			setTimeout(() => setWritingLabel(''), 400);
+		} else {
+			setIsWritingIsDefined(true);
+			setWritingLabel(isWritingLabel);
+		}
+	}, [isWritingLabel]);
+
+	return (
+		<Container width="70%" crossAlignment="flex-start">
+			{
```
Comment (lisaparma) [inline src/meetings/components/sidebar/MeetingConversationAccordion/MeetingChatAccordionTitle.tsx]: `{` and `}` are useless if there isn't a condition

**Comment by lisaparma** [comment]
If the last message is a configuration message, when the "is typing" label appears there is a small change in height

https://github.com/zextras/carbonio-ws-collaboration-ui/assets/30258876/1c1eb20d-8245-4f81-bc3d-7fcf35ade660

**Review by lisaparma — CHANGES_REQUESTED** [review]

**Review by lisaparma — APPROVED** [review]

**Review by CerDennis — APPROVED** [review]

**Pull request merged**